.github/workflows/gradle-build.yml

```diff
@@ -0,0 +1,61 @@
+# This workflow will build a Java project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Java CI with Gradle
+
+on:
+  pull_request:
+    branches: [ dev ]
+    paths:
+      - 'src/**'
+      - '.github/**'
+      - '!.gradle/wrapper'
+      - '!.gitignore'
+      - '!LICENSE'
+      - '!THIRD PARTY NOTICES'
+      - '!*.md'
+      - '*.gradle'
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK
+        uses: actions/setup-java@v2
+        with:
+          java-version: '16'
+          distribution: 'adopt'
+          cache: gradle
+      - name: Detect Secrets
+        uses: RobertFischer/detect-secrets-action@v2.0.0
+      - run: .\scripts\decodeAndWrite.ps1 -encodedValue $env:ENCODED_VALUE -outputPath $env:OUTPUT_PATH
```

Comment:
I don't think you need to download any of those files just for the build

Reply:
We did it in the previous ado pipelines and we do it in the current core workflow also. Do we not need them in this case because we have moved to source-promotion?

Reply:
ignore this comment, thinking about the details made me realize we do need them, even the snapshots are signed.

Reply:
actually sorry, this two workflows structure is confusing me a little.
We don't need to download those here as those file contain the passwords to upload to sonatype and sign-in credentials. This workflow doesn't publish anything so it doesn't need either.
The other workflow however needs the secring.
The reason why it wasn't there in the previous version is because this was done at the build step with build promotion, but we now switched to source promotion.